### DISCUSSION TITLE
Make open source?

### DISCUSSION THREAD

**Discussion by weisJ** [Anze/KeyCluCask#22] [Q&A]
Is there a plan to make the source code openly available?

**Comment by Anze** [comment]
At the moment - no. But at some point it would be open source.

**Comment by leikoilja** [comment]
our of curiosity, why not open source?

**Comment by Anze** [comment]
I chose not to make it open source to maintain control over its development and ensure a consistent user experience.
This approach enables me to ensure the software's quality, and focus on delivering a polished and well-maintained product while still keeping it accessible to everyone.
However, I would make it open source when I would become unable to continue its development.

**Reply by mkoslacz** [reply]
I don't quite understand this reasoning. Even with open-source code, you would maintain control over development as the sole developer and have full authority over which pull requests to accept. The only changes would be that the code becomes publicly visible and others could fork it. Given these points, I don't see any conflict with your goals.

**Reply by drowld** [reply]
Yeah it's a bit of a silly argument, now i want to make a fork to add features i want for my custom system and i basically have to redo everything ?
It's free so i'm not sure what's the idea of not open sourcing it :)
Would love if you could enlighten us!!

Also full mac access without open source mehhh

**Comment by acgourley** [comment]
This is entirely your right but I caution anyone to _**avoid**_ installing a closed source application which asks for accessibility permissions as a matter of basic security practices, no matter how handy it is. It's _almost_ full access to your mac.

**Comment by Blacks-Army** [comment]
I hope you may reconsider your decision someday.

The arguments you mentioned don’t really convince me. If the app is already free, I don’t fully understand why keeping it closed source is necessary, especially when transparency is important for user trust.

You mentioned that you would open source it if you were no longer able to continue development, but from a user’s perspective, making it open source while you are still actively maintaining it could help improve trust, quality, and the long-term future of the project.

Thank you for your work, and I hope you might reconsider in the future.

**Comment by xJaywalk** [comment]
Maybe I am sensitive. Seeing the answer from owner I just delete it immediately.  I didn't notice this repo is not really open source at first.

**Reply by leikoilja** [reply]
yeh, same tbh. discovered this app 3 years ago, and the needs is still there but not planning to install it unless it's actually open sourced
